addons/calendar/data/mail_template_data.xml

```diff
@@ -91,7 +91,7 @@
                     </li>
                 </t>
                 <t t-if="recurrent">
-                    <li>When: <t t-out="object.recurrence_id.name or ''">Every 1 Weeks, for 3 events</t></li>
+                    <li>When: <t t-out="object.recurrence_id.compute_name(lang)">Every 1 Weeks, for 3 events</t></li>
```

Comment:
since the method is only called on a single record, you could do `.get_recurrence_name()`, there doesn't seem to be a need to the compute_name method